x/oracle/client/rest/query.go

```diff
@@ -92,7 +136,65 @@ func getOracleScriptByIDHandler(clientCtx client.Context) http.HandlerFunc {
 		}
 
 		vars := mux.Vars(r)
-		res, height, err := clientCtx.Query(fmt.Sprintf("custom/%s/%s/%s", oracletypes.QuerierRoute, oracletypes.QueryOracleScripts, vars[idTag]))
+		res, height, err := clientCtx.Query(fmt.Sprintf(
+			"custom/%s/%s/%s",
+			oracletypes.QuerierRoute,
+			oracletypes.QueryOracleScript,
+			vars[idTag],
+		))
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		clientCtx = clientCtx.WithHeight(height)
+		rest.PostProcessResponse(w, clientCtx, res)
+	}
+}
+
+func getOracleScriptsHandler(clientCtx client.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		clientCtx, ok := rest.ParseQueryHeightOrReturnBadRequest(w, clientCtx, r)
+		if !ok {
+			return
+		}
+
+		body, err := ioutil.ReadAll(r.Body)
```

Comment:
Cosmos sdk have method for this github.com/cosmos/cosmos-sdk/types/rest
rest.ReadRESTReq(w, r, clientCtx.LegacyAmino, &req)
(check it and replace, please everywhere)